test/run_spineopt_investments.jl

```diff
@@ -95,6 +95,7 @@ function _test_capacity_investments()
         url_in, url_out, file_path_out = _test_run_spineopt_investments_setup()
         object_parameter_values = [
             ["model", "instance", "use_connection_intact_flow", false],
+            ["unit", "unit_a", "units_on_cost", 0], # Just to have units_on variables
```

Comment:
Nice catch @nnhjy !

In my opinion this is good to merge, the only question is why this. Why do we need it?

Reply:
Ah, I overlooked the case where the investment mode requires `units_on` to bound investment transition. I will fix this. Thanks for pointing this out.